tests/acceptance/00_basics/04_bundles/unless_bug.cf

```diff
@@ -0,0 +1,44 @@
+body common control
+{
+    inputs => { "../../default.cf.sub" };
+    bundlesequence => { default($(this.promise_filename)) };
+}
+
+bundle agent test
+{
+  meta:
+    any::
+      "description" -> { "CFE-2698" }
+        string => "This test shows a bug in unless which should be fixed";
+
+  vars:
+    any::
+      "target_variable"
+        string => "value";
+
+      "vars_promise_which_should_be_skipped"
+        string => "value",
+        unless => isvariable( $(variable_defined_later) );
+
+      # If you move this variable up, the test will pass,
+      # the order shouldn't matter here, this is a bug
+      "variable_defined_later"
+        string => "target_variable";
+}
+
+bundle agent check
+{
+  classes:
+    any::
+      "not_ok"
+        or => {
+            isvariable("test.vars_promise_which_should_be_skipped"),
+        };
+
+  reports:
+    # TOOD: Swap these when bug is fixed:
+    not_ok::
+      "$(this.promise_filename) Pass";
+    !not_ok::
+      "$(this.promise_filename) FAIL";
```

Comment:
This should use the `test_soft_fail` mechanism.

Reply:
https://github.com/cfengine/core/pull/3852